internal/plugin/installer/http_installer_test.go

```diff
@@ -348,3 +348,255 @@ func TestMediaTypeToExtension(t *testing.T) {
 		}
 	}
 }
+
+func TestExtractWithNestedDirectories(t *testing.T) {
+	source := "https://repo.localdomain/plugins/nested-plugin-0.0.1.tar.gz"
+	tempDir := t.TempDir()
+
+	// Set the umask to default open permissions so we can actually test
+	oldmask := syscall.Umask(0000)
```

Comment:
I'm inclined to say that this should explicitly open files with the permissions it needs during Open calls.  If you call `syscall.Umask` , I'm wondering if you hit Ctrl-C Ctrl-C in an unfortunate timing, your shell might get stuck creating new files with 777 permissions unless you explicitly close the terminal and restart it. Or for subprocesses of this test case to not know that the umask has changed.

that might be overly paranoid though.